src/main/java/com/laytonsmith/core/AliasCore.java

```diff
@@ -344,121 +326,218 @@ public final void reload(MCPlayer player, String[] settings, boolean firstLoad)
-										ConfigRuntimeException.HandleUncaughtException(
-												new ConfigCompileException("Duplicate command defined. (First occurance found at "
-												+ s1.getTarget() + ")", s2.getTarget()), "Duplicate command.", player);
-									}
-								}
-							}
-						}
-					}
+					Globals.clear();
 				} finally {
-					compilerMSA.stop();
+					clearingGlobals.stop();
 				}
 			}
-			// Register commands, and set up tabcompleters and things
-			for(int i = 0; i < scripts.size(); i++) {
-				Script s1 = scripts.get(i);
-				if(!s1.getSmartComment().getAnnotations("@command").isEmpty()) {
-					registerCommand(s1, env);
+
+			// Run extension's onShutdown()
+			if(options.reloadExtensions()) {
+				ProfilePoint extManagerShutdown = profiler.start("Extension manager shutdown", LogLevel.VERBOSE);
+				try {
+					ExtensionManager.Shutdown();
+				} finally {
+					extManagerShutdown.stop();
 				}
 			}
-		} catch (IOException ex) {
-			Static.getLogger().log(Level.SEVERE, "Path to config file is not correct/accessible."
-					+ " Please check the location and try loading the plugin again.");
-		} catch (Throwable t) {
-			t.printStackTrace();
 		}
 
-		ProfilePoint postReloadAliases = parent.profiler.start("Extension manager post reload aliases", LogLevel.VERBOSE);
+		// Reload things here that have their own reload option and also run on startup.
+		if(options.reloadExtensions()) {
+			ProfilePoint extensionManagerStartup = profiler.start("Extension manager startup", LogLevel.VERBOSE);
+			try {
+				ExtensionManager.Startup();
+			} finally {
+				extensionManagerStartup.stop();
+			}
+		}
+
+		// Everything else should be reloaded now, so execute successfully compiled scripts and register commands
+		if(options.reloadScripts() && env != null) {
+			ProfilePoint executeMS = profiler.start("Execution of MS files in Local Packages", LogLevel.VERBOSE);
+			try {
+				env.getEnv(GlobalEnv.class).SetLabel(Static.GLOBAL_PERMISSION);
+				env.getEnv(CommandHelperEnvironment.class).SetCommandSender(Static.getServer().getConsole());
+				localPackages.executeMS(env);
+			} finally {
+				env.getEnv(CommandHelperEnvironment.class).SetCommandSender(null);
+				executeMS.stop();
+			}
+
+			ProfilePoint registerCommands = profiler.start("Registering of annotated commands", LogLevel.VERBOSE);
+			try {
+				// Register commands, and set up tabcompleters and things
+				for(Script s1 : scripts) {
+					if(!s1.getSmartComment().getAnnotations("@command").isEmpty()) {
+						registerCommand(s1, env);
+					}
+				}
+			} finally {
+				registerCommands.stop();
+			}
+		}
+
+		// Allow extensions know we are done reloading. Runs extension's onPostReloadAliases().
+		ProfilePoint postReloadAliases = profiler.start("Extension manager post reload aliases", LogLevel.VERBOSE);
 		try {
 			ExtensionManager.PostReloadAliases();
 		} finally {
 			postReloadAliases.stop();
 		}
+
+		String output;
+		if(firstLoad) {
+			output = "Loading complete. ";
```

Comment:
This is a bit picky, but I'd personally prefer either `Load` and `Reload` or `Loading` and `Reloading` in these two cases of feedback.